pyoozie/tags.py

```diff
@@ -494,6 +496,133 @@ def _xml(self, doc, tag, text):
         return doc
 
 
+class _WorkflowEntity(typing.Iterable):
+    """An abstract object representing an Oozie workflow action that can be serialized to XML."""
+    # pylint: disable=abstract-method
+
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(
+            self,
+            xml_tag,       # typing.Text
+            name=None,     # type: typing.Optional[typing.Text]
+            on_error=None  # type: typing.Optional[_WorkflowEntity]
+    ):
+        # type: (...) -> None
+        self.xml_tag = xml_tag
+        self.__name = name
+        self.__on_error = copy.deepcopy(on_error)
+        self.__uid = uuid.uuid4().hex[:8]
```

Comment:
As `__uid` is only used during creation of the identifier, we can assign `__name` as - 
```self.__name = name or uuid.uuid4().hex[:8]```. This will help make the `create_identifier` method succinct.